### ISSUE TITLE
automatic meta element

### ISSUE THREAD

**Issue by goatshriek** [goatshriek/stumpless#224]
[RFC 5424](https://datatracker.ietf.org/doc/html/rfc5424#section-7.3) specifies that a `meta` structured data element should be included to give some level of information about the log message itself. Stumpless should have the ability to generate this element automatically, as well as the capability to include it automatically when logging to given targets.

**Comment by junyeong0619** [comment]
<PRI>VERSION TIMESTAMP HOSTNAME APP-NAME PROCID MSGID [STRUCTURED-DATA] MSG  
You mean that stumpless should save log file like this when each time that vstumpless_add_log make log?

**Comment by goatshriek** [comment]
Yes, this is the only format currently supported by stumpless, the RFC 5424 format. This issue is to support a new element in the structured data that generates the fields described in section 7.3 automatically.

To make this more extensible, I'll probably implement this by adding a visitor function to the targets, similar to the way each target has a filter function. This will then be used to create and add the element to the entry before it is logged. This will also be used to implement #223 and #225.

This will require a lot of work with core implementation details, which is why none of these issues is marked as a good first issue right now. If you're interested in taking a crack at them, I can implement the visitor in the next few weeks so that the implementation is more accessible.

**Comment by goatshriek** [comment]
Interested parties can watch the `target-refactor` branch to see this work in progress.

**Comment by junyeong0619** [comment]
While you're working to make this issue more accessible, I will be analyzing the target-refactor branch in the meantime. I truly appreciate your consistently kind and helpful responses
